syntax = "proto3";
package mediheaven;

service Code {
    rpc getCode (CodeRequest) returns (CodeResponse);
    rpc delCode (CodeDelRequest) returns (SuccessResponse);
    rpc listCode (CodeListRequest) returns (CodeListResponse);
}

service Account {
    rpc register (RegisterRequest) returns (SuccessResponse);
    rpc Login (LoginRequest) returns (LoginResponse);
}


/*Service Code*/
message CodeRequest {
    string email = 1;
    string account_type = 4;
    string timestamp = 5;
    string signature = 2;
    string expiration_date = 3;
}

message CodeResponse {
    bool successful = 1;
    string code = 2;
    optional string msg = 3;
}

message CodeDelRequest {
    string email = 1;
    string code = 2;
    string timestamp = 3;
    string signature = 4;
}

message CodeListRequest {
    string email = 1;
    string timestamp = 2;
    string signature = 3;
}

message CodeListResponse {
    repeated CodeMessage codes = 1;
    optional string msg = 2;
}

message CodeMessage {
    string account_type = 1;
    string code = 2;
    string expiration_date = 3;
}

/*Service Account*/
message RegisterRequest {
    string Code = 1;
    string Account_Type = 2;
    string First_Name = 3;
    string Last_Name = 4;
    string Sex = 5;
    int32 Age = 6;
    string Date_Of_Birth = 7;
    string Phone_Number = 8;
    string Email = 9;
    string Pub_key = 10;
}

message SuccessResponse {
    bool successful = 1;
    optional string msg = 2;
}

message LoginRequest {
    string email = 1;
    string timestamp = 4;
    string signature = 2;
    string account_type = 3;
}

message LoginResponse {
    bool successful = 1;
    string pub_key = 2;
    optional string msg = 3;
}